templates/_components/add-to-calendar.twig

```diff
@@ -0,0 +1,72 @@
+{%
+set classes = [
+  "localgov-add-to-calendar"
+]
+%}
+
+{{ attach_library('localgov_base/add-to-calendar') }}
+
+{% set lgdAddToCalendarAttributes = create_attribute() %}
+
+{% set ical_link %}data:text/calendar;charset=utf8,BEGIN:VCALENDAR%0AVERSION:2.0%0ABEGIN:VEVENT{% if atc_start_date %}%0ADTSTART:{{ atc_start_date|date("Ymd\\THi00\\Z") }}{% endif %}{% if atc_end_date %}%0ADTEND:{{ atc_end_date|date("Ymd\\THi00\\Z") }}{% endif %}{% if atc_title %}%0ASUMMARY:{{ atc_title }}{% endif %}{% if atc_details %}%0ADESCRIPTION: {{ atc_details|replace({'<br />': '\n', '<br>': '\\n', '<p>': '', '</p>': '\\n\\n'}) }}{% endif %}{% if atc_location %}%0ALOCATION: {{ atc_location|replace({'<br />': ' ', '<br>': ' ', '<p>': ' ', '</p>': ''})}}{% endif %}{% if atc_rrule %} %0ARRULE: {{ atc_rrule }} {% endif %}%0AEND:VEVENT%0AEND:VCALENDAR{% endset %}
+
+{% set google_link %}
+    https://www.google.com/calendar/r/eventedit?text={{ atc_title }}
```

Comment:
This assembly of URLs would be better done in PHP code rather than in twig.

I think a more flexible solution would be to build this as a formatter alterer for the date field.

Reply:
It looks like this contrib module does what I mean - https://www.drupal.org/project/addtocal_augment. Though I don't know what SmartDate is -- it looks like a different ecosystem from Date Recur module that we use, so probably not suitable for us. 

The same approach would work though -- although it should be simpler than that one, as I don't see what Date Augmenter API is needed, since Drupal core provides the means to alter field formatters.

Reply:
> This assembly of URLs would be better done in PHP code rather than in twig.

Do you want to amend the PR to do this?

Reply:
@joachim-n thanks for your comments, just discussing in Merge Tuesday, are you planning to do some work on this?

Reply:
So having stared at this for a bit.

As it stands, realistically, just moving the URL assembly to PHP would be moving the construction of some of the output from the twig template to the preprocess. 

What joachim also points to would be the more ideal of having this add to the display widget itself, the additional bonus there would be that it can be enabled / disabled. But there isn't really a convenient way of adding to the Formatter Plugin itself (what would it be a hook_form_alter, and also a template override?). Making the formatter pluggable itself to add features would be a huge bonus that could be contributed back but feels like a few hours work?

Reply:
I will emphasis - as it stands - as there is no config to switch this on or off it will appear on every site inheriting the base theme, and to remove it (no I don't know why people would want to do that) requires extending the the template to remove what is added to the event full template.

Reply:
We could add a new boolean field to the event content type to say "add Add to Calendar" widget to this event" and have it switched on by default for all new events. I'm not sure if that's overkill, or if that a "next phase of work" job for us.

Reply:
@ekes how about we merge this as is, then later

1. If someone wants to move the construction of the URL from twig to PHP they can do that in a follow up issue.
2. If we think it's a good idea to have a checkbox on each event to "Turn off add to calendar for this event" we can create a follow-up issue for this (in the localgov_events repo)

Reply:
Yep fine with me.